demo/docbot/cluster_bootstrap.py

```diff
@@ -1,131 +1,228 @@
-  #       "model": "embed-english-v2.0",
-  #       "truncate": "END"
-  #     },
-  #    "actions": [{
-  #        "action_type": "predict",
-  #        "method": "POST",
-  #        "url": "https://api.cohere.ai/v1/embed",
-  #        "headers": {
-  #                "Authorization": "Bearer ${credential.cohere_key}"
-  #            },
-  # 			"request_body": "{ \"texts\": ${parameters.prompt}, \"truncate\": \"${parameters.truncate}\", \"model\": \"${parameters.model}\" }",
-  # 			"pre_process_function": "connector.pre_process.cohere.embedding",
-  # 			 "post_process_function": "connector.post_process.cohere.embedding"
-  #        }]
-  # }
-  pass
-
-
-def initialize_model():
-  #   POST /_plugins/_ml/models/_register?deploy=true
-  # {
-  #     "name": "embed-english-v2.0",
-  #     "function_name": "remote",
-  #     "description": "test model",
-  #     "model_group_id": "<MODEL_GROUP_ID>",
-  #     "connector_id": "<CONNECTOR_ID>"
-  # }
-  pass
-
-
-def initialize_ingestion_pipeline():
-  #   PUT _ingest/pipeline/cohere-ingest-pipeline
-  # {
-  #   "description": "Cohere Neural Search Pipeline",
-  #   "processors" : [
-  #     {
-  #       "text_embedding": {
-  #         "model_id": "<MODEL_ID>",
-  #         "field_map": {
-  #           "content": "content_embedding"
-  #         }
-  #       }
-  #     }
-  #   ]
-  # }
-
-  pass
-
-
-def initialize_index():
-  # PUT /cohere-index
-  # {
-  # 	"settings": {
-  # 		"index.knn": true,
-  # 		"default_pipeline": "cohere-ingest-pipeline"
-  # 	},
-  # 	"mappings": {
-  # 		"properties": {
-  # 			"content_embedding": {
-  # 				"type": "knn_vector",
-  # 				"dimension": 4096,
-  # 				"method": {
-  # 					"name": "hnsw",
-  # 					"space_type": "cosinesimil",
-  # 					"engine": "nmslib"
-  # 				}
-  # 			},
-  # 			"content": {
-  # 				"type": "text"
-  # 			}
-  # 		}
-  # 	}
-  # }
-
-  pass
+import requests
+import os, json, sys
+from util import opensearch_connection_builder
+sys.path.append('./demo/')
+from docbot.util import opensearch_connection_builder
+
+import dotenv
+from os import getenv
+
+dotenv.load_dotenv()
+COHERE_KEY = getenv('COHERE_KEY')
+MODEL_STATE = {
+    "model_group_id": "",
+    "connector_id": "",
+    "model_id": ""
+}
+
+
+
+def initialize_cluster_settings(client):
+    current_settings = client.cluster.get_settings()
+
+    data = {
+        "plugins.ml_commons.allow_registering_model_via_url": True,
+        "plugins.ml_commons.only_run_on_ml_node": False,
+        "plugins.ml_commons.connector_access_control_enabled": True,
+        "plugins.ml_commons.model_access_control_enabled": True,
+        "plugins.ml_commons.trusted_connector_endpoints_regex": [
+            "^https://runtime\\.sagemaker\\..*[a-z0-9-]\\.amazonaws\\.com/.*$",
+            "^https://api\\.openai\\.com/.*$",
+            "^https://api\\.cohere\\.ai/.*$"
+        ]
+    }
+
+    if 'persistent' in current_settings and current_settings['persistent'] == data:
```

Comment:
`current_settings['persistent'] == data` this will fail as it doesn't do deep object inspection. This comparison could be put into a util function that deeply compares the objects.

Reply:
Getting handled in #63